server/index.js

```diff
@@ -0,0 +1,29 @@
+const WebSocket = require("ws");
+const rl = require("../lib/rl");
+const { runPlayLoop } = require("../lib/play");
+
+console.log("*****************");
+console.log("STARTING");
+const wss = new WebSocket.Server({ port: 8080 });
+
+wss.on("connection", async (ws) => {
+  console.log("New client connected!");
+
+  ws.on("close", () => {
+    console.log("Client has disconnected");
+  });
+
+  // UGLY AF HAX
+  const question = (prompt, cb) => {
```

Comment:
I was thinking about how we could make this cleaner:

Similar to how you have store.log() (which either uses console.log or `this.ws.send()`), you could make an async function `await store.ask(prompt)`, which either uses `rl.question()` or `ws.send()` and `ws.onmessage()`.

I had a look, and you're only using rl.question in 4 places, so this might be an easy-enough refactor
```
$ git grep -F .question | grep -v test
lib/chooseNumPlayers.js:5:    rl.question("Set the number of human players. 1 or 2? = ", (choice) => {
lib/chooseTeam.js:8:    rl.question(question, (choice) => {
lib/chooseUserGrid.js:5:    rl.question("Enter your choice of grid number = ", (choice) => {
lib/playAgain.js:5:    rl.question("Play again? (y/n) = ", (answer) => {
```

You also have 16 places in tests where you do `rl.question = (question, cb) => { cb(input) }`. You could probably replace them with `store.ask = async (prompt) => input` I think.

Reply:
The refactor that we are doing, to move shared global things like `rl.question` and `console.log` into the `store` object is a classic object-oriented refactoring technique (I can't find a name for it though).